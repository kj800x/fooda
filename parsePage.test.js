const fs = require("fs");

const { parsePage } = require("./parsePage");

describe("parsePage", () => {
  const cases = fs.readdirSync("./test_cases");
  for (var i = 0; i < cases.length; i++) {
    const testCase = cases[i];
    it(`works on ${testCase}`, () => {
      expect(
        parsePage(fs.readFileSync(`./test_cases/${testCase}`))
      ).toMatchSnapshot();
    });
  }
});
