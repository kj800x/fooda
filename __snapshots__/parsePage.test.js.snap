// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsePage works on today_active.html 1`] = `
Object {
  "currentDay": 27,
  "menuDay": 27,
  "popups": Array [
    Object {
      "link": "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=6563",
      "name": "Jersey Mike's",
    },
    Object {
      "link": "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=4785",
      "name": "Brothers Restaurant Tacos",
    },
  ],
}
`;

exports[`parsePage works on today_inactive.html 1`] = `
Object {
  "currentDay": 27,
  "menuDay": 28,
  "popups": Array [
    Object {
      "link": "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=6563",
      "name": "Jersey Mike's",
    },
    Object {
      "link": "http://app.fooda.com/accounts/1729/popup/menu_page/P0733968/items?filterable%5Bvendor_id%5D=4785",
      "name": "Brothers Restaurant Tacos",
    },
  ],
}
`;
